pandas/core/arrays/arrow/accessors.py

```diff
@@ -0,0 +1,182 @@
+            raise AttributeError(self._validation_msg)
+
+    @property
+    def dtypes(self) -> Series:
+        """
+        Return the dtype object of each child field of the struct.
+
+        Returns
+        -------
+        pandas.Series
+            The data type of each child field.
+
+        Examples
+        --------
+        >>> import pyarrow as pa
+        >>> s = pd.Series(
+        ...     [
+        ...         {"version": 1, "project": "pandas"},
+        ...         {"version": 2, "project": "pandas"},
+        ...         {"version": 1, "project": "numpy"},
+        ...     ],
+        ...     dtype=pd.ArrowDtype(pa.struct(
+        ...         [("version", pa.int64()), ("project", pa.string())]
+        ...     ))
+        ... )
+        >>> s.struct.dtypes
+        version     int64[pyarrow]
+        project    string[pyarrow]
+        dtype: object
+        """
+        from pandas import (
+            Index,
+            Series,
+        )
+
+        pa_type = self._parent.dtype.pyarrow_dtype
+        types = [ArrowDtype(pa_type[i].type) for i in range(pa_type.num_fields)]
+        names = [pa_type[i].name for i in range(pa_type.num_fields)]
+        return Series(types, index=Index(names))
+
+    def field(self, name_or_index: str | int) -> Series:
+        """
+        Extract a child field of a struct as a Series.
+
+        Parameters
+        ----------
+        name_or_index : str | int
+            Name or index of the child field to extract.
+
+        Returns
+        -------
+        pandas.Series
+            The data corresponding to the selected child field.
+
+        See Also
+        --------
+        Series.struct.to_frame : Return all child fields as a DataFrame.
+
+        Examples
+        --------
+        >>> import pyarrow as pa
+        >>> s = pd.Series(
+        ...     [
+        ...         {"version": 1, "project": "pandas"},
+        ...         {"version": 2, "project": "pandas"},
+        ...         {"version": 1, "project": "numpy"},
+        ...     ],
+        ...     dtype=pd.ArrowDtype(pa.struct(
+        ...         [("version", pa.int64()), ("project", pa.string())]
+        ...     ))
+        ... )
+
+        Extract by field name.
+
+        >>> s.struct.field("project")
+        0    pandas
+        1    pandas
+        2     numpy
+        Name: project, dtype: string[pyarrow]
+
+        Extract by field index.
+
+        >>> s.struct.field(0)
+        0    1
+        1    2
+        2    1
+        Name: version, dtype: int64[pyarrow]
+        """
+        from pandas import Series
+
+        pa_arr = self._parent.array._pa_array
+        if isinstance(name_or_index, int):
+            index = name_or_index
+        else:
+            index = pa_arr.type.get_field_index(name_or_index)
+
+        pa_field = pa_arr.type[index]
+        field_arr = pc.struct_field(pa_arr, [index])
+        return Series(field_arr, dtype=ArrowDtype(field_arr.type), name=pa_field.name)
```

Comment:
This needs to be updated to preserve the index of `self._parent`. Also, add tests for a non-default index.

Reply:
Done